### ISSUE TITLE
Set-PSResourceRepository the 'ApiVersion' parameter raises an exception if the value 'Unknown' is specified

### ISSUE THREAD

**Issue by LaurentDardenne** [PowerShell/PSResourceGet#1347]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

```Powershell
Set-PSResourceRepository PSGallery -ApiVersion 'unknown'
```

Note:
This call works : 
```Powershell
Set-PSResourceRepository PSGallery -ApiVersion 'unknown' -Trusted
# or Set-PSResourceRepository PSGallery -ApiVersion 'unknown' -Priority 50

Get-psResourceRepository |select name,apiversion

Name        ApiVersion
----        ----------
PSGallery      unknown
PSRepoLocal    unknown
```

In the following case the error message indicates the value 'unknow' as allowed...
```Powershell
C:\temp> set-psResourceRepository PSRepoLocal -ApiVersion 'Test'
Set-PSResourceRepository : Impossible de lier le paramètre «ApiVersion». Impossible de convertir la valeur «Test» en
type «Microsoft.PowerShell.PSResourceGet.UtilClasses.PSRepositoryInfo+APIVersion». Erreur: «Impossible de faire
correspondre le nom d'identificateur Test à un nom d'énumérateur valide. Spécifiez l'un des noms d'énumérateur
suivants et réessayez :
unknown, v2, v3, local, nugetServer »
```

### Expected behavior

```console
?
The documentation is not up to date so I don't know if it's a bug or not.

```
According to this [code](https://github.com/PowerShell/PSResourceGet/blob/931115343784bdc1c716c46c99d9ecf2767c01f5/src/code/RepositorySettings.cs#L209) it seems to be correct but this confuses me :-)

### Actual behavior

```console
> set-psResourceRepository PSGallery -ApiVersion 'unknown'
set-psResourceRepository : Must set Uri, Priority, Trusted, ApiVersion, or CredentialInfo parameter
Au caractère Ligne:1 : 1
+ set-psResourceRepository PSGallery -ApiVersion 'unknown'
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument : (Microsoft.Power...ourceRepository:SetPSResourceRepository) [Set-PSRes
   ourceRepository], PSInvalidOperationException
    + FullyQualifiedErrorId : ErrorInNameParameterSet,Microsoft.PowerShell.PSResourceGet.Cmdlets.SetPSResourceRepository
```

### Error details

```console
$Error[0]|select *


PSMessageDetails      :
Exception             : System.Management.Automation.PSInvalidOperationException: Must set Uri, Priority, Trusted,
                        ApiVersion, or CredentialInfo parameter
                           à System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord
                        errorRecord)
TargetObject          : Microsoft.PowerShell.PSResourceGet.Cmdlets.SetPSResourceRepository
CategoryInfo          : InvalidArgument : (Microsoft.Power...ourceRepository:SetPSResourceRepository)
                        [Set-PSResourceRepository], PSInvalidOperationException
FullyQualifiedErrorId : ErrorInNameParameterSet,Microsoft.PowerShell.PSResourceGet.Cmdlets.SetPSResourceRepository
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : à <ScriptBlock>, <Aucun fichier> : ligne 1
PipelineIterationInfo : {}
```

### Environment data

```powershell
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Binary     0.5.23     Microsoft.PowerShell.PSResourceGet  {Find-PSResource, Get-InstalledPSResource, Get-PSResourceR...

Key   : PSVersion
Value : 5.1.19041.3031
Name  : PSVersion


Key   : PSEdition
Value : Desktop
Name  : PSEdition


Key   : PSCompatibleVersions
Value : {1.0, 2.0, 3.0, 4.0, 5.0, 5.1.19041.3031}
Name  : PSCompatibleVersions


Key   : BuildVersion
Value : 10.0.19041.3031
Name  : BuildVersion


Key   : CLRVersion
Value : 4.0.30319.42000
Name  : CLRVersion


Key   : WSManStackVersion
Value : 3.0
Name  : WSManStackVersion


Key   : PSRemotingProtocolVersion
Value : 2.3
Name  : PSRemotingProtocolVersion


Key   : SerializationVersion
Value : 1.1.0.1
Name  : SerializationVersion
```

**Comment by SydneyhSmith** [comment]
Thanks @LaurentDardenne this is resolved with PR #1310 and is available in the latest beta (beta24)...I will open an issue to improve the documentation for this parameter

**Comment by LaurentDardenne** [comment]
>  this is resolved 

Unless it's a documentation issue, I don't see any improvement :

![image](https://github.com/PowerShell/PSResourceGet/assets/4894452/cd7ff213-6745-4999-ae08-a8103945cfce)
